cmake/modules/CompileWarnings.cmake

```diff
@@ -0,0 +1,8 @@
+macro(wpilib_target_warnings target)
+    if(NOT MSVC)
+        target_compile_options(${target} PRIVATE -Wall -pedantic -Wextra -Werror -Wno-unused-parameter)
+    else()
+        target_compile_options(${target} PRIVATE /wd4244 /wd4267 /wd4146)
+        target_compile_options(${target} PRIVATE /WX /wd4996 /permissive- /Zc:throwingNew)
```

Comment:
Hmm. Not sure if we should remove `/permissive-` from all downstream users. Unlike the other ones, code that compiles with `permissive-` might not compile without the flag, and vise versa. It actually changes a TON of things about how the compiler works.

Reply:
After reading more about it, it sounds more like the /std flag.  I'll move it (and /Zc:throwingNew) back into the wpiutil global flags.